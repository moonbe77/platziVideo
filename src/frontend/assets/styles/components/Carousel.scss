.carousel {
  padding: 0 5px;
  margin: 20px 0;
  width: 100%;
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  flex-shrink: unset;
  justify-content: space-between;
}

.carousel__container {
  // border: 2px solid red;
  transition: 450ms -webkit-transform;
  transition: 450ms transform;
  transition: 450ms transform, 450ms -webkit-transform;
  font-size: 0;
  white-space: nowrap;
  padding-bottom: 10px;
  width: 100vw;
  overflow: hidden;
}

.carousel__container:hover .carousel-item {
  opacity: 0.3;
}

.carousel__container:hover .carousel-item:hover {
  -webkit-transform: scale(1.1);
  transform: scale(1.1);
  opacity: 1;
}

.carousel__content {
  // border: 2px solid blue;
  margin: 35px 0px;
  position: relative;
  transition: transform 1s cubic-bezier(0.17, 0.07, 0, 1.07) 0s;
}

.carousel__slide__button {
  width: 5vw;
  z-index: 999;
  position: absolute;
  align-items: center;
  margin: auto 0;
  height: 250px;
  display: flex;
  justify-content: center;
}

.action-left {
  top: 35px;
  left: 0px;
  background: rgb(0, 0, 0);
  background: linear-gradient(
    90deg,
    rgba(0, 0, 0, 1) 0%,
    rgba(251, 251, 251, 0) 90%
  );
}

.action-right {
  top: 35px;
  right: 0px;
  background: rgb(0, 0, 0);
  background: linear-gradient(
    270deg,
    rgba(0, 0, 0, 1) 0%,
    rgba(251, 251, 251, 0) 90%
  );
}
