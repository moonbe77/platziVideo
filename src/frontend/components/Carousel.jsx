/* eslint-disable react/jsx-closing-bracket-location */
/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';
import '../assets/styles/components/Carousel.scss';

const Carousel = ({ children }) => {
  const [mousePosition, setMousePosition] = useState(0);
  const [carrouselPosition, setCarrouselPosition] = useState(0);

  useEffect(() => {
    const carouselNode = document.querySelector('.carousel');
    const carouselContainerNode = document.querySelector(
      '.carousel__container',
    );
    const carrouselBoxWidth = carouselNode.offsetWidth;
    const carrouselContainerTotalWidth = carouselContainerNode.scrollWidth;
    let interval;

    if (mousePosition.x <= 100 && mousePosition.x >= 20) {
      console.log('seting interval');
      if (carrouselPosition !== 0) {
        interval = setInterval(() => {
          setCarrouselPosition(carrouselPosition => carrouselPosition + 1);
        }, 2000);
      }
    } else {
      console.log('cleaning interval');
      clearInterval(interval);
    }

    if (mousePosition.x >= carrouselBoxWidth - 100) {
      setCarrouselPosition(carrouselPosition - 1);
      // setInterval(() => {
      // }, 2000);
    }
  });

  // eslint-disable-next-line arrow-parens
  const handleSlideCarousel = direction => {
    if (direction === 'left') {
      setCarrouselPosition(carrouselPosition - 200);
    } else {
      setCarrouselPosition(carrouselPosition + 200);
    }
  };

  return (
    // eslint-disable-next-line jsx-a11y/mouse-events-have-key-events
    <section
      className='carousel'
      // onMouseMove={e => setMousePosition({ x: e.clientX, y: e.clientY })}
      // onMouseOut={() => setMousePosition({ x: 0, y: 0 })}
    >
      <div
        className='carousel__slide__button action-left'
        role='button'
        tabIndex='0'
        onClick={() => handleSlideCarousel('left')}>
        left
      </div>
      <div className='carousel__container'>
        <div className='carousel__content' style={{ transform: `translate(${carrouselPosition}px, 0px)` }}>
          {children}
        </div>
      </div>
      <div
        className='carousel__slide__button action-right'
        role='button'
        tabIndex='0'
        onClick={() => handleSlideCarousel('right')}>
        right
      </div>
    </section>
  );
};

export default Carousel;
