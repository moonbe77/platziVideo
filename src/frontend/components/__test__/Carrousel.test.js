import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/MockProvider';
import Carousel from '../Carousel';

describe('Carousel Component', () => {
  test('match Snapshot', () => {
    const carousel = create(
      <ProviderMock>
        <Carousel />
      </ProviderMock>,
    );
    expect(carousel.toJSON()).toMatchSnapshot();
  });
});
