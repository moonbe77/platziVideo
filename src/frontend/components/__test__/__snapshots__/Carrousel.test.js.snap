// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component match Snapshot 1`] = `
<section
  className="carousel"
>
  <div
    className="carousel__container"
  />
</section>
`;
